draco/core/misc.py

```diff
@@ -167,7 +167,7 @@ def process(self, tstream, gain):
         if weight_arr is not None:
 
             # Convert dynamic range to a binary weight and apply to data
-            gain_weight = (weight_arr[:] > 2.0).astype(np.float64)
+            gain_weight = (weight_arr[:] > 0.0).astype(np.float64)
             tstream.weight[:] *= gain_weight[:, np.newaxis, :]
```

Comment:
I think this needs the input axis to be cast into a product axis and then the mask applied to the timestream.